internal/node/controller/controller2.go

```diff
@@ -0,0 +1,101 @@
+package controller
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/eagraf/habitat-new/core/state/node"
+	"github.com/eagraf/habitat-new/internal/node/hdb"
+	"github.com/eagraf/habitat-new/internal/process"
+	"github.com/pkg/errors"
+)
+
+type controller2 struct {
+	db             hdb.Client
+	processManager process.ProcessManager
+}
+
+func newController2(pm process.ProcessManager, db hdb.Client) (*controller2, error) {
+	// Validate types of all input components
+	_, ok := pm.(node.Component[process.RestoreInfo])
+	if !ok {
+		return nil, fmt.Errorf("Process manager of type %T does not implement Component[*node.Process]", pm)
+	}
+
+	ctrl := &controller2{
+		processManager: pm,
+		db:             db,
+	}
+
+	return ctrl, nil
+}
+
+func (c *controller2) getNodeState() (node.State, error) {
+	var nodeState node.State
+	err := json.Unmarshal(c.db.Bytes(), &nodeState)
+	if err != nil {
+		return node.State{}, err
+	}
+	return nodeState, nil
+}
+
+func (c *controller2) startProcess(installationID string) error {
+	state, err := c.getNodeState()
+	if err != nil {
+		return err
+	}
+	app, ok := state.AppInstallations[installationID]
+	if !ok {
+		return fmt.Errorf("app with ID %s not found", installationID)
+	}
+
+	bytes, err := state.Bytes()
+	if err != nil {
+		return errors.Wrap(err, "error getting state.Bytes()")
+	}
+
+	transition, err := node.GenProcessStartTransition(installationID, bytes)
```

Comment:
Why do we pass it in as bytes here? Seems like we immediate unmarshal it here: https://github.com/eagraf/habitat-new/pull/40/files#diff-59e76c44f0435924ed3f11fcc8d86189c2ea0d032a5656f2e066ef4dd4d430bdR377

Reply:
Oops, will remove that extra work.